src/include/sof/preproc.h

```diff
@@ -35,20 +35,21 @@
  * works for predefined number of args
  * PP_NARG_BEFORE_COMPILE(A,B,C,D) evaluates to 4
  */
-#define PP_NARG_BEFORE_COMPILE(...) \
-		PP_NARG_BEFORE_COMPILE_(_, ##__VA_ARGS__, PP_RSEQ_N())
-#define PP_NARG_BEFORE_COMPILE_(_, ...) \
+#define PP_NARG_BEFORE_COMPILE(...) IF_ELSE(HAS_ARGS(__VA_ARGS__)) (\
+		PP_NARG_BEFORE_COMPILE_(__VA_ARGS__, PP_RSEQ_N()) )\
+		(0)
+#define PP_NARG_BEFORE_COMPILE_(...) \
 		PP_ARG_N(__VA_ARGS__)
-#define PP_ARG_N( _0, \
-	 _1,  _2,  _3,  _4,  _5,  _6,  _7,  _8,  _9, _10, \
+#define PP_ARG_N(\
+	_1,   _2,  _3,  _4,  _5,  _6,  _7,  _8,  _9, _10, \
```

Comment:
@mwierzbix this looks like both patches can be squashed too since similar changes in each patch ?

Reply:
I'll squash them, right now I am researching that Travis build breakage, but once I figure it out I'll squash everything.

Reply:
Squashed.